pkg/controllers/management/auth/legacy_rt_cleaner.go

```diff
@@ -8,14 +8,12 @@ import (
 )
 
 type rtCleaner struct {
-	clusterLister v3.ClusterLister
-	mgmt          *config.ManagementContext
+	mgmt *config.ManagementContext
```

Comment:
I assume the thought here is that the mgmt context already has a clusterLister?
Just a question, feel free to disagree...
Would we be better off having just the roleTemplate client and clusterLister here instead of the whole mgmt context.